sdk/tracing/azotel/go.mod

```diff
@@ -1,8 +1,8 @@
 module github.com/Azure/azure-sdk-for-go/sdk/tracing/azotel
 
-go 1.22.7
+go 1.23.0
 
-toolchain go1.23.1
+toolchain go1.24.0
```

Comment:
Hmm I wonder if we should set the `toolchain` to the same version as the Go version. Thoughts?

Reply:
Or I wonder if we even need to specify this? I've been reading https://go.dev/doc/toolchain but I haven't completely wrapped my head around it yet.

Reply:
Me neither 😆 

`go get` seems always to add it whereas `mod tidy` does so only for some states. I don't want it if it isn't necessary, so I suppose I'll push the "mod tidy doesn't want to add a toolchain directive" state